scripts/test/lib_subctl_recover_brokerinfo_tests.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+# shellcheck disable=SC2154
+
+function join_cluster() {
+    echo "Joining cluster: $1"
+    _subctl join broker-info.subm --kubeconfig "${KUBECONFIGS_DIR}"/kind-config-"$1"
+}
+
+function uninstall_subm() {
+    echo "Uninstalling submariner from $1 cluster"
+    _subctl uninstall -y --kubeconfig "${KUBECONFIGS_DIR}"/kind-config-"$1"
+}
+
+function setContext() {
+    echo "Setting context to $1"
+    kubectl config use-context "$1"
+}
+
+function filesmatch() {
+    echo "Checking if the original and newly created broker-info.subm matches"
+    cmp broker-info.subm.orig broker-info.subm || return 1
+}
+
+function use_context() {
+  echo "using context" "$1"
+  kubectl config use-context "$1"
+}
+
+function no_broker_found_test() { # Blue
+  echo "Testing the scenarios where Broker is not installed on the specified context"
+  for cluster1subm in y n; do
+    for cluster2subm in y n; do
+      for context in cluster1 cluster2; do
```

Comment:
I know I suggested the use of loops like this initially, but only if it helped clarify things. The end result here doesn’t benefit from the use of loops, at least not using the structure shown in the PR.

(As an example, it may ultimately be more useful to loop over an array of clusters; but the best way to go about this is to start with a sequential list of test commands, and see from that whether there’s opportunity and relevance in factoring code.)